halo2_proofs/src/poly/kzg/commitment.rs

```diff
@@ -88,11 +88,9 @@ where
         };
 
         let mut g_lagrange_projective = vec![E::G1::identity(); n as usize];
-        let mut root = E::Scalar::ROOT_OF_UNITY_INV.invert().unwrap();
-        for _ in k..E::Scalar::S {
-            root = root.square();
-        }
-        let n_inv = Option::<E::Scalar>::from(E::Scalar::from(n).invert())
+        let root = E::Scalar::ROOT_OF_UNITY.pow_vartime([1 << (E::Scalar::S - k)]);
```

Comment:
I think we could keep the original for-loop-squaring, it should be faster than using `pow_vartime`.

```suggestion
        let mut root = E::Scalar::ROOT_OF_UNITY;
        for _ in k..E::Scalar::S {
             root = root.square();
         }
```

Reply:
@han0110 I do benchmark on my own computer before i do PR. The result was attached in the PR's comment.

With `k=24`:
- `for_loop_squaring` cost: 14.104 µs
- `pow_vartime` cost: 1.7520 µs

Cost reduced with `pow_vartime` is: -89.897%

Reply:
The benchmark:

```rust
        let k = 24;
        use halo2curves::bn256::Bn256;
        use halo2curves::pairing::Engine;

        c.bench_function("root-of-unity", |b| {
            b.iter(|| {
                let mut root = <Bn256 as Engine>::Scalar::ROOT_OF_UNITY_INV
                    .invert()
                    .unwrap();
                for _ in k..<Bn256 as Engine>::Scalar::S {
                    root = root.square();
                }
            })
        });

        c.bench_function("root-of-unity 2", |b| {
            b.iter(|| {
                let root = <Bn256 as Engine>::Scalar::ROOT_OF_UNITY
                    .pow_vartime([1 << (<Bn256 as Engine>::Scalar::S - k)]);
            })
        });
```

Reply:
Oh, I see. The problem is `.invert()`

Reply:
Ah I think the thing that reduces cost is removal of `invert`.

We get the `root` by inverting `root_inv`, it worked but slowly. It could be a small typo when changing the code.